File: q0094/q0094.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution
{
public:
    vector<int> inorderTraversal(TreeNode *root)
    {
        vector<int> vals;
        dfsHelper(root, vals);
        return vals;
    }

private:
    void dfsHelper(const TreeNode *node, vector<int> &vals)
    {
        if (node == nullptr)
            return;
        dfsHelper(node->left, vals);
        vals.push_back(node->val);
        dfsHelper(node->right, vals);
    }
};

File: q0094/q0094.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        dfsHelper(root, res);
        return res;
    }

    private void dfsHelper(TreeNode node, List<Integer> vals) {
        if (node == null) {
            return;
        }
        dfsHelper(node.left, vals);
        vals.add(node.val);
        dfsHelper(node.right, vals);
    }
}